src/kakarot/instructions/system_operations.cairo

```diff
@@ -556,31 +669,68 @@ namespace CreateHelper {
         pedersen_ptr: HashBuiltin*,
         range_check_ptr,
         bitwise_ptr: BitwiseBuiltin*,
-    }(
-        ctx: model.ExecutionContext*, value: felt, offset: felt, size: felt, salt: felt
-    ) -> model.ExecutionContext* {
+    }(ctx: model.ExecutionContext*, popped_len: felt, popped: Uint256*) -> model.ExecutionContext* {
         alloc_locals;
-        let (evm_contract_address, starknet_contract_address) = ContractAccount.deploy(salt);
 
         // Load bytecode code from memory
+        let value = popped[0];
+        let offset = popped[1];
+        let size: Uint256 = popped[2];
+
         let (bytecode: felt*) = alloc();
         let (memory, gas_cost) = Memory.load_n(
-            self=ctx.memory, element_len=size, element=bytecode, offset=offset
+            self=ctx.memory, element_len=size.low, element=bytecode, offset=offset.low
         );
         let ctx = ExecutionContext.update_memory(ctx, memory);
 
         let ctx = ExecutionContext.increment_gas_used(
             self=ctx, inc_value=gas_cost + SystemOperations.GAS_COST_CREATE
         );
 
+        // create2 context pops 4 off the stack, create pops 3
+        // so we use popped_len to derive the way we should handle
+        // the creation of evm addresses
+        local evm_contract_address;
+
+        if (popped_len != 4) {
```

Comment:
let's do this `if` at the very end to return in both branches and avoid all the `tempvar` (Lucas Levy advises me to do this one day)